spec/javascripts/components/story/expanded_story/expanded_story_state_spec.js

```diff
@@ -128,89 +144,95 @@ describe('<ExpandedStoryState />', () => {
     it('triggers the edit callback passing the story state', () => {
       const state = states[0];
       const change = states[1];
-
-      const story = {
-        _editing: { state },
+      const onEditSpy = vi.fn();
+      const props = {
+        story: {
+          _editing: { state },
+        },
+        onEdit: onEditSpy,
+        disabled: false,
       };
 
-      const onEdit = vi.fn();
-
-      const wrapper = shallow(
-        <ExpandedStoryState story={story} onEdit={onEdit} disabled={false} />
-      );
-      const select = wrapper.find('select');
-
-      select.simulate('change', { target: { value: change } });
+      const { container } = renderComponent(props);
+      const select = container.querySelector('select');
+      fireEvent.change(select, { target: { value: change } });
 
-      expect(onEdit).toHaveBeenCalledWith(change);
+      expect(onEditSpy).toHaveBeenCalledWith(change);
     });
   });
 
   describe('when component is disabled', () => {
     it('select field is editable', () => {
       const onEditSpy = vi.fn();
-      const story = { state: 'started', _editing: { state: 'started' } };
-      const wrapper = mount(
-        <ExpandedStoryState story={story} onEdit={onEditSpy} disabled={true} />
-      );
-      const select = wrapper.find('select');
-      expect(select.prop('disabled')).toBe(true);
+      const props = {
+        story: { state: 'started', _editing: { state: 'started' } },
+        onEdit: onEditSpy,
+        disabled: true,
+      };
+
+      const { container } = renderComponent(props);
+      const select = container.querySelector('select');
+
+      expect(select).toBeDisabled();
```

Comment:
Shouldn't the select be editable?

Reply:
Yeah that's puzzling, but the pre-migration test was doing exactly that:

```js
describe('when component is disabled', () => {
    it('select field is editable', () => {
      const onEditSpy = vi.fn();
      const story = { state: 'started', _editing: { state: 'started' } };
      const wrapper = mount(
        <ExpandedStoryState story={story} onEdit={onEditSpy} disabled={true} />
      );
      const select = wrapper.find('select');
      expect(select.prop('disabled')).toBe(true);
    });
  });
```

So I just "translated" it to testing-library. Should we change it, then?

Reply:
See. Then, we could replace the test title to "select field is disabled". Makes more sense I guess. What do you think?

Reply:
Done!

[5db207f](https://github.com/Codeminer42/cm42-central/pull/926/commits/5db207f92631cbd842057dd074abf7f0721c356c)